### PR TITLE
Pass to libdrm render node, not primary node

### PR DESCRIPTION
**Pull request by Alexandr-Konovalov**
Fix failure on CentOS 7.5: primary drm node was created for libdri,
but it lacks permissions required for ioctl(.., DRM_IOCTL_I915_GETPARAM)
to run.

### CODE REVIEW

**Comment by alariono** [comment]
@Alexandr-Konovalov,
add link on issue https://github.com/Intel-Media-SDK/MediaSDK/issues/674 into commit message|
GitHub will close issue automatically once the commit is merged into the default branch.

File: samples/sample_common/src/vaapi_device.cpp [hunk 1]
```diff
@@ -95,7 +95,9 @@ mfxStatus CVAAPIDeviceX11::Init(mfxHDL hWindow, mfxU16 nViews, mfxU32 nAdapterNu
     }
     m_xcbconn = x11xcblib.XGetXCBConnection(m_dpy);
 
-    m_dri_fd = open("/dev/dri/card0", O_RDWR);
+    // it's enough to pass render node, because we only request
+    // information from kernel via m_dri_fd
+    m_dri_fd = open("/dev/dri/renderD128", O_RDWR);
```
Comment (dvrogozh) [inline samples/sample_common/src/vaapi_device.cpp]: this looks ok, I think.

However, the rest of the function does not make sense to me. For example, on line 67 we call XCreateSimpleWindow, thus we should already have X display opened, but we open it on the line 91 actually. Just grepping the code:
```
samples/sample_common/src/vaapi_device.cpp:91:    m_dpy = x11lib.XOpenDisplay(NULL);
samples/sample_common/src/vaapi_utils_x11.cpp:49:            m_display = m_x11lib.XOpenDisplay(currentDisplay);
samples/sample_common/src/vaapi_utils_x11.cpp:51:            m_display = m_x11lib.XOpenDisplay(VAAPI_X_DEFAULT_DISPLAY);
samples/sample_common/src/vaapi_utils_x11.cpp:136:            m_display = m_x11lib.XOpenDisplay(currentDisplay);
samples/sample_common/src/vaapi_utils_x11.cpp:138:            m_display = m_x11lib.XOpenDisplay(VAAPI_X_DEFAULT_DISPLAY);
```
Do we really need all 3 functions?

File: samples/sample_common/src/vaapi_utils_x11.cpp [hunk 2]
```diff
@@ -51,114 +50,99 @@ X11LibVA::X11LibVA(void)
...
-            va_res = m_libva.vaInitialize(m_va_dpy, &major_version, &minor_version);
-            sts = va_to_mfx_status(va_res);
-            if (MFX_ERR_NONE != sts) {
-                m_x11lib.XCloseDisplay(m_display);
-                }
+                va_res = m_libva.vaInitialize(m_va_dpy, &major_version, &minor_version);
+                sts = va_to_mfx_status(va_res);
                 unsetenv("LIBVA_DRIVER_NAME");
             } else {
                 sts = MFX_ERR_NOT_INITIALIZED;
```
Comment (Alexandr-Konovalov) [inline samples/sample_common/src/vaapi_utils_x11.cpp]: I treated lack of m_x11lib.XCloseDisplay(m_display) on that branch as a bug. Is it correct or there was some plan?

File: samples/sample_common/src/vaapi_device.cpp (hunk 1 above)
Comment (Alexandr-Konovalov) [reply samples/sample_common/src/vaapi_device.cpp]: It seems CVAAPIDeviceX11 can use connection from m_X11LibVA it includes.

File: samples/sample_common/src/vaapi_utils_x11.cpp [hunk 3]
```diff
@@ -51,114 +50,99 @@ X11LibVA::X11LibVA(void)
...
-            {
-                m_libva.vaTerminate(m_va_dpy);
-                m_x11lib.XCloseDisplay(m_display);
-                sts = MFX_ERR_NULL_PTR;
-            }
-            if (!setenv("LIBVA_DRIVER_NAME", RENDERING_DRIVER_NAME, 1)) {
-                va_res = m_libva.vaInitialize(m_va_dpy_render, &major_version, &minor_version);
-                sts = va_to_mfx_status(va_res);
-                if (MFX_ERR_NONE != sts)
+            if (!msdk_strcmp(PROCESSING_DRIVER_NAME, RENDERING_DRIVER_NAME)) {
```
Comment (dvrogozh) [inline samples/sample_common/src/vaapi_utils_x11.cpp]: I believe we need to completely drop this processing/rendering drivers stuff. So, the behind the scenes story was that iHD did not support efficient on-screen rendering via X11, but i965 did. So, to optimized that we have used 2 drivers approach. However, now we are going to support iHD driver only, so I believe that all this dual driver approach could be dropped. @alariono : do you agree?

This should be a preceding patch I believe.

Comment (alariono) [reply samples/sample_common/src/vaapi_utils_x11.cpp]: @dvrogozh, Yes, agree lets drop dual driver approach

Comment (Alexandr-Konovalov) [reply samples/sample_common/src/vaapi_utils_x11.cpp]: Dropped, please check.

File: samples/sample_common/src/vaapi_utils_x11.cpp [hunk 4]
```diff
@@ -29,8 +29,7 @@ or https://software.intel.com/en-us/media-client-solutions-support.
 
 #define VAAPI_X_DEFAULT_DISPLAY ":0.0"
 
-const char* PROCESSING_DRIVER_NAME="iHD";
-const char* RENDERING_DRIVER_NAME="iHD";
+const char* DRIVER_NAME="iHD";
```
Comment (dvrogozh) [inline samples/sample_common/src/vaapi_utils_x11.cpp]: drop.

File: samples/sample_common/src/vaapi_utils_x11.cpp [hunk 5]
```diff
@@ -64,11 +63,11 @@ X11LibVA::X11LibVA(void)
         }
         if (MFX_ERR_NONE == sts)
         {
-            if (!setenv("LIBVA_DRIVER_NAME", PROCESSING_DRIVER_NAME, 1)) {
-            va_res = m_libva.vaInitialize(m_va_dpy, &major_version, &minor_version);
-            sts = va_to_mfx_status(va_res);
-            if (MFX_ERR_NONE != sts) {
-                m_x11lib.XCloseDisplay(m_display);
+            if (!setenv("LIBVA_DRIVER_NAME", DRIVER_NAME, 1)) {
```
Comment (dvrogozh) [inline samples/sample_common/src/vaapi_utils_x11.cpp]: nope, you don't need to set _any_ driver name. If needed user will do that in a shell. Just call vaInitialize().

File: samples/sample_common/src/vaapi_utils_x11.cpp [hunk 6]
```diff
@@ -169,14 +147,11 @@ X11LibVA::X11LibVA(void)
...
+    //release context
+    if (m_contextID != VA_INVALID_ID)
     {
-        //release context
-        if (m_contextID != VA_INVALID_ID)
-        {
-            m_libva.vaDestroyContext(m_va_dpy_render, m_contextID);
-            m_contextID = VA_INVALID_ID;
-        }
+        m_libva.vaDestroyContext(m_va_dpy_render, m_contextID);
```
Comment (dvrogozh) [inline samples/sample_common/src/vaapi_utils_x11.cpp]: What is m_va_dpy_render if you did not create it? I believe you need to delete this variable as well and use m_va_dpy everywhere.

File: samples/sample_common/src/vaapi_utils_x11.cpp [hunk 7]
```diff
@@ -50,14 +49,23 @@ X11LibVA::X11LibVA(void)
...
+
+            if (!m_va_dpy)
+            {
+                sts = MFX_ERR_NULL_PTR;
+            }
+        }
+#if !defined(X11_DRI3_SUPPORT)
+        if (MFX_ERR_NONE == sts)
+        {
             m_va_dpy_render = m_vax11lib.vaGetDisplay(m_display);
```
Comment (dvrogozh) [inline samples/sample_common/src/vaapi_utils_x11.cpp]: remove and use m_va_dpy

File: samples/sample_common/src/vaapi_utils_x11.cpp (hunk 6 above)
Comment (Alexandr-Konovalov) [reply samples/sample_common/src/vaapi_utils_x11.cpp]: Done, thank you.

**Comment by dvrogozh** [comment]
@alariono : looks good to me, your turn.

**Pull request merged**